### ISSUE TITLE
Discard Changes doesn't actually happen until second try

### ISSUE THREAD

**Issue by theowiersema** [microsoft/vscode#157314]
When undoing changes tracked via git inside of VSCode, I have to select "Discard Changes" twice before the changes actually disappear from the list in the "Source Control" sidebar. 

![image](https://user-images.githubusercontent.com/7004529/183147414-b31f1e78-ce08-4cb7-b440-d62c961fcfa5.png)

![image](https://user-images.githubusercontent.com/7004529/183144511-9b80e4c0-27bc-4e3b-b045-40a5998b2202.png)

**Comment by lszomoru** [comment]
@theowiersema, are you able to consistently reproduce this behaviour when discarding changes? If so, could you please enable `trace` logging for the git extension as described in [this wiki page](https://github.com/microsoft/vscode/wiki/Git-&-SCM-Issues#logging)? After that, reload the window, try to reproduce the problem. Please only invoke the "Discard Changes" command once. After that please share the contents of [the git output window](https://code.visualstudio.com/docs/editor/versioncontrol#_git-output-window). Thank you!

**Comment by theowiersema** [comment]
Yeah, it happesn 100% of the time for me. Here's the git output:

```
[2022-08-10T13:14:44.877Z] > git checkout -q -- C:\sourcecode\client\src\components\ProcessInstance\index.jsx [214ms]
[2022-08-10T13:14:44.985Z] [debug] [wt] Change: c:\sourcecode\client\.git\index
[2022-08-10T13:14:44.985Z] [debug] [wt] Change: c:\sourcecode\client\src\components\ProcessInstance\index.jsx
[2022-08-10T13:14:44.985Z] [debug] [.git] Change: c:\sourcecode\client\.git\index
[2022-08-10T13:14:45.082Z] > git status -z -uall [197ms]
[2022-08-10T13:14:45.369Z] > git symbolic-ref --short HEAD [272ms]
[2022-08-10T13:14:45.803Z] > git for-each-ref --format=%(refname)%00%(upstream:short)%00%(objectname)%00%(upstream:track)%00%(upstream:remotename)%00%(upstream:remoteref) refs/heads/autosaveprocesses refs/remotes/autosaveprocesses [405ms]
[2022-08-10T13:14:46.276Z] > git remote --verbose [439ms]
[2022-08-10T13:14:46.525Z] > git ls-files --stage -- C:\sourcecode\client\src\components\ProcessInstance\index.jsx [408ms]
[2022-08-10T13:14:46.544Z] > git for-each-ref --sort -committerdate --format %(refname) %(objectname) %(*objectname) [726ms]
[2022-08-10T13:14:46.858Z] > git cat-file -s c47930693856f1e0d5b14aaabd852b65d9aa3ab7 [318ms]
[2022-08-10T13:14:46.919Z] > git config --get commit.template [355ms]
[2022-08-10T13:14:47.125Z] > git show --textconv :src/components/ProcessInstance/index.jsx [241ms]
```

**Comment by theowiersema** [comment]
@lszomoru Anything else I can try?

**Comment by lszomoru** [comment]
@theowiersema, those outputs looks identical to me so it's not clear to me what is going on. Couple of things to try:
1. Are you able to reproduce the problem if you disable all extensions (ex: start code with `--disable-extensions`)
2. Can you try the following: 
    - Make a change to a particular file
    - Click on the "Discard Changes" button
    - Click on the "Refresh" button and see if the item is gone

Providing a recording of the problem (https://gifcap.dev) might also help. Thank you!

**Comment by theowiersema** [comment]
@lszomoru It still happens when all extensions are disabled. I've attached a GIF of it happening. 
![Recording 2022-09-08 at 11 09 09](https://user-images.githubusercontent.com/7004529/189172151-4016f48d-2cad-4622-8f1e-6fab32542a6d.gif)

**Comment by theowiersema** [comment]
Hey @lszomoru looks like this is fixed in 1.73.1! I can't reproduce it anymore.
